run_deterministic_tests.py

```diff
@@ -0,0 +1,337 @@
+        args, test_strategies, strats_created_from_block
+    )
+
+    # Write the strategy txhashs to a json file
+    test_manager.write_strategy_txhashs_to_json(test_strategy_txhashs)
+
+    # Run the results crosscheck task
+    run_results_crosscheck_task(args, proc)
+
+
+def run_results_crosscheck_task(args, proc: subprocess.Popen):
+    """
+    Run the results crosscheck task.
+
+    Args:
+        args: argparse.Namespace, the command line arguments
+        proc: subprocess.Popen, the process
+    """
+    args.logger.info("\nRunning run_results_crosscheck_task...")
+
+    # Initialize the tx helper
+    tx_helper = TestTxHelper()
+
+    # Wait for the transactions to be completed
+    test_strategy_txhashs = tx_helper.wait_for_txs(args)
+
+    # Scan for successful transactions on Tenderly which are marked by status=1
+    actual_txs = tx_helper.tx_scanner(args)
+    expected_txs = tx_helper.load_json_file("test_results.json", args)
+
+    results_description = tx_helper.log_results(
+        args, actual_txs, expected_txs, test_strategy_txhashs
+    )
+    proc.terminate()
+    for k, v in results_description.items():
+        args.logger.info(f"{k}: {v}")
+
+
+def main(args: argparse.Namespace):
+    """
+    Main function for the script. Runs the specified task based on the command line arguments.
+
+    Args:
+        args: argparse.Namespace, the command line arguments
+    """
+
+    # Set up the logger
+    args.logger = get_logger(args)
+    args.logger.info(f"Running task: {args.task}")
+
+    # Set the timeout in seconds
+    args.timeout = args.timeout_minutes * 60
+
+    if str(args.create_new_testnet).lower() == "true":
+        uri, from_block = TestManager.create_new_testnet()
+        args.rpc_url = uri
+
+    # Initialize the Web3 Manager
+    test_manager = TestManager(args=args)
+    if args.delete_old_logs:
+        test_manager.delete_old_logs(args)
+
+    if args.task == "set_test_state":
+        set_test_state_task(test_manager.w3)
+    elif args.task == "get_carbon_strategies_and_delete":
+        get_carbon_strategies_and_delete_task(test_manager, args)
+    elif args.task == "run_tests_on_mode":
+        _extracted_task_handling(test_manager, args)
+    elif args.task == "end_to_end":
+        get_carbon_strategies_and_delete_task(test_manager, args)
+        set_test_state_task(test_manager.w3)
+        _extracted_task_handling(test_manager, args)
+    else:
+        raise ValueError(f"Task {args.task} not recognized")
+
+
+def _extracted_task_handling(test_manager: TestManager, args: argparse.Namespace):
+    """
+    Extracted task handling.
+    """
+    test_strategies = test_manager.get_test_strategies(args)
+    run_tests_on_mode_task(args, test_manager, test_strategies)
+
+
+if __name__ == "__main__":
+    # Parse the command line arguments
+    parser = argparse.ArgumentParser(description="Fastlane Bot")
+    parser.add_argument(
+        "--task",
+        default="update_pool_params",
```

Comment:
This default value is not listed in the choices, and does not appear to be handled.